### ISSUE TITLE
snap tries to bootstrap via local resolver since v0.108.0-b.50

### ISSUE THREAD

**Issue by jkreileder** [AdguardTeam/AdGuardHome#6457]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, ARM64

### Installation

Snapcraft

### Setup

On one machine

### AdGuard Home version

v0.108.0-b.50

### Action

Since v0.108.0-b.50 adguard home seems to ignore bootstrap servers and instead tries to use the local resolver for bootstrapping.  When the local resolver is configured to use adguard home, this of course fails.

Setup:
- adguard home snap running on eth0 (192.168.78.56)
- host running systemd-resolve on lo (127.0.0.53) configured to forward to 192.168.78.56 (ie. adguard) via DHCP
- DNS servers:
```
[/fritz.box/]192.168.78.1
https://dns.google/dns-query
https://dns11.quad9.net/dns-query
```
- Bootstrap servers:
```
192.168.78.1
8.8.8.8
8.8.4.4
2001:4860:4860::8888
2001:4860:4860::8844
9.9.9.11
149.112.112.11
2620:fe::11
2620:fe::fe:11
```
- Private reverse DNS server: 192.168.78.1

Adguard startup:
```
10:10:54.151674 eth0  Out IP 192.168.78.56.60361 > 192.168.78.1.53: 13727+ PTR? 56.78.168.192.in-addr.arpa. (44)
10:10:54.152145 eth0  In  IP 192.168.78.1.53 > 192.168.78.56.60361: 13727* 1/1/3 PTR ubuntu.fritz.box. (160)
```

As soon as adguard receives a query, I see requests for bootstrapping going to systemd-resolved which doesn't work as it tries to go back to adguard in this setup. I'd expect these request to go to one of the configured bootstrap servers.

```
10:11:17.272527 lo    In  IP 192.168.78.56.33264 > 192.168.78.56.53: 47122+ A? xxxxxxxxxxxxxxx.com. (37)
10:11:17.274919 lo    In  IP 127.0.0.1.47302 > 127.0.0.53.53: 12587+ [1au] A? dns.google. (39)
10:11:17.274919 lo    In  IP 127.0.0.1.39047 > 127.0.0.53.53: 63644+ [1au] AAAA? dns11.quad9.net. (44)
10:11:17.274919 lo    In  IP 127.0.0.1.57320 > 127.0.0.53.53: 2105+ [1au] AAAA? dns.google. (39)
10:11:17.274927 lo    In  IP 127.0.0.1.56440 > 127.0.0.53.53: 3413+ [1au] A? dns11.quad9.net. (44)
10:11:17.275587 lo    In  IP 192.168.78.56.51281 > 192.168.78.56.53: 34665+ [1au] AAAA? dns11.quad9.net. (44)
10:11:17.276029 lo    In  IP 192.168.78.56.46958 > 192.168.78.56.53: 12488+ [1au] AAAA? dns.google. (39)
10:11:17.276492 lo    In  IP 192.168.78.56.46741 > 192.168.78.56.53: 56941+ [1au] A? dns.google. (39)
10:11:17.277014 lo    In  IP 192.168.78.56.59331 > 192.168.78.56.53: 29915+ [1au] A? dns11.quad9.net. (44)
```

Going back to b.49 immediately solves the problem.  (So does giving resolved another DNS server - but adguard b.50 still skips the configured bootstrap servers in this case)

The adguard logs (even in verbose mode) don't contain anything useful.

### Expected result

Bootstrap servers getting used instead of local resolver.

### Actual result

Local resolver is used instead of bootstrap servers.

**Comment by deezeem** [comment]
I can reproduce as well.  All my configured DoH upstream DNS servers fail, moved all upstreams to IP's only as a work around.

**Comment by gyCfjSnO** [comment]
Upon upgrading to version 0.108.0-b.50, I encountered the same problem where it abruptly stopped functioning. I observed that AdGuard was attempting to query itself, which is an unusual behavior that I hadn't seen before. Until this issue is addressed, I am compelled to use the router's DNS as a workaround.

**Comment by jtp10181** [comment]
I ran into the same issue using all DoT servers. DNS totally stopped working after updating. I did not have time to mess with it so I just rolled back from the automatic backup. I assume it was this same issue.

**Comment by EdiTurn** [comment]
Same issue here, and now using [Specifying upstreams for domains](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains) as workaround.

**Comment by EugeneOne1** [comment]
@jkreileder, @deezeem, @gyCfjSnO, @jtp10181, @EdiTurn, hello and apologies for the late response.  We've found the bug and pushed the fix to the `edge` update channel, so it should now bootstrap as intended.  Could you please try it out?

**Comment by jkreileder** [comment]
@EugeneOne1 v0.108.0-a.777+849abaf2?
That version is still broken for me with the setup described above.

From the commit log, I guess I should wait for 0e4e243?!

**Comment by bestpika** [comment]
> @EugeneOne1 v0.108.0-a.777+849abaf2? That version is still broken for me with the setup described above.
> 
> From the commit log, I guess I should wait for [0e4e243](https://github.com/AdguardTeam/AdGuardHome/commit/0e4e243e8e21e74a1af506a6653dd0c92ee7376d)?!

I am currently using v0.108.0-a.778+0e4e243e.
Looks like the problem should be resolved.

**Comment by EugeneOne1** [comment]
@jkreileder, yeah, you're absolutely right.  I was under the impression the 0e4e243 has already been released on the edge.
